### ISSUE TITLE
no-undef false positive 

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#463]
```
...

  function createHttpClient(options) {
    return new HttpClient(opCreateHttpClient(options));
  }

  function opCreateHttpClient(args) {
    return core.jsonOpSync("op_create_http_client", args);
  }

  class HttpClient {
    constructor(rid) {
      this.rid = rid;
    }
    close() {
      core.close(this.rid);
    }
  }


...
```

```
error[no-undef]: HttpClient is not defined
   --> /Users/biwanczuk/dev/deno/op_crates/fetch/26_fetch.js:877:16
    |
877 |     return new HttpClient(opCreateHttpClient(options));
    |                ^^^^^^^^^^
    |
```

Note: the file in question is a script, but it shouldn't really matter here.

**Comment by magurotuna** [comment]
I'm looking into it.

**Comment by magurotuna** [comment]
A Minimal reproducible example is:

```ts
(() => {
  function foo() {
    return new Bar();
  }
  class Bar {}
})();
```

Currently `no-undef` rule only checks variables in the global scope, though non-global variables (including class declaration) should also be checked as the above example shows.
To achive it, it seems like the analyzer similar to the one that is used in `prefer-const` rule is suitable. I'll fix this issue in this way.
